sky/clouds/service_catalog/gcp_catalog.py

```diff
@@ -364,68 +363,119 @@ def list_accelerators(
-                                                        use_spot=True,
-                                                        region=region_filter,
-                                                        zone=None)
-            new_infos[info.accelerator_name].append(
-                info._replace(
-                    instance_type=vm_type,
-                    cpu_count=cpu_count,
-                    memory=memory,
-                    # total cost = VM instance + GPU.
-                    price=info.price + vm_price,
-                    spot_price=info.spot_price + vm_spot_price,
-                ))
-    return new_infos
+
+    # This is a simplified version of get_instance_type_for_accelerator, as it
+    # has to be applied to all the entries in acc_host_df, and the input is more
+    # restricted.
+    def _get_host_vm(acc_name: str, acc_count: int, region: str,
+                     zone: str) -> Optional[str]:
+        df = _df[(_df['Region'].str.lower() == region.lower()) &
+                 (_df['AvailabilityZone'].str.lower() == zone.lower()) &
+                 (_df['InstanceType'].notna())]
+        if acc_name in _ACC_INSTANCE_TYPE_DICTS:
+            instance_types = _ACC_INSTANCE_TYPE_DICTS[acc_name][acc_count]
+            df = df[df['InstanceType'].isin(instance_types)]
+        else:
+            if acc_name not in _NUM_ACC_TO_NUM_CPU:
+                acc_name = 'DEFAULT'
+
+            cpus = _NUM_ACC_TO_NUM_CPU[acc_name][acc_count]
+            # The memory size should be at least 4x the requested number of
+            # vCPUs to be consistent with all other clouds.
+            memory = cpus * _DEFAULT_GPU_MEMORY_CPU_RATIO
+            df = df[df['InstanceType'].str.startswith(_DEFAULT_HOST_VM_FAMILY)]
+            df = df[(df['vCPUs'] >= cpus) & (df['MemoryGiB'] >= memory)]
+        df = df.dropna(subset=['Price'])
+        # Some regions may not have the host VM available, although the GPU is
+        # offered, e.g., a2-megagpu-16g in asia-northeast1.
+        if df.empty:
+            return None
+        row = df.loc[df['Price'].idxmin()]
+        return row['InstanceType']
+
+    acc_host_df = _df
+    tpu_df = None
+    # If price information is not required, we do not need to fetch the host VM
+    # information.
+    if require_price:
+        acc_host_df = _df[_df['AcceleratorName'].notna()]
+        # Filter the acc_host_df first before fetching the host VM information.
+        # This is to reduce the rows to be processed for optimization.
+        # TODO: keep it sync with `list_accelerators_impl`
+        if gpus_only:
+            acc_host_df = acc_host_df[~acc_host_df['GpuInfo'].isna()]
+        if name_filter is not None:
+            acc_host_df = acc_host_df[
+                acc_host_df['AcceleratorName'].str.contains(
+                    name_filter, case=case_sensitive)]
+        if region_filter is not None:
+            acc_host_df = acc_host_df[acc_host_df['Region'] == region_filter]
```

Comment:
This should be `contains` instead of `==`?

https://github.com/skypilot-org/skypilot/blob/c304eec7c4036b62a61e9590347f47534af6d76a/sky/clouds/service_catalog/common.py#L495-L498

Reply:
Good catch! Replaced with the impl in the common.py